diff --git a/extras/source/glade/libreoffice-catalog.xml.in b/extras/source/glade/libreoffice-catalog.xml.in
index cb5b695f86da..5bc09b4ddddb 100644
--- a/extras/source/glade/libreoffice-catalog.xml.in
+++ b/extras/source/glade/libreoffice-catalog.xml.in
@@ -51,7 +51,7 @@
                         generic-name="Template Icon View" parent="GtkIconView"
                         icon-name="widget-gtk-iconview"/>
 
-    <glade-widget-class title="DataTreeListBox" name="svxcorelo-DataTreeListBox"
+    <glade-widget-class title="DataTreeListBox" name="svxcorezh-DataTreeListBox"
                         generic-name="DataTreeListBox" parent="GtkTreeView"
                         icon-name="widget-gtk-treeview"/>
 
