diff --git a/include/sal/config.h b/include/sal/config.h
index 335c4174e3ae..408b5a78b6b7 100644
--- a/include/sal/config.h
+++ b/include/sal/config.h
@@ -20,6 +20,8 @@
 #ifndef INCLUDED_SAL_CONFIG_H
 #define INCLUDED_SAL_CONFIG_H
 
+#define DLL_FNAME_POSTFIX  "zh"
+
 #if defined LIBO_INTERNAL_ONLY && defined ANDROID && defined __cplusplus
 #include <android/compatibility.hxx>
 #endif
diff --git a/include/tools/svlibrary.h b/include/tools/svlibrary.h
index 2a46d8c88412..c06bd431552c 100644
--- a/include/tools/svlibrary.h
+++ b/include/tools/svlibrary.h
@@ -8,7 +8,8 @@
  */
 
 #ifndef SVLIBRARY
-#define SVLIBRARY( Base ) SAL_MODULENAME( Base "lo" )
+#define DLL_FNAME_POSTFIX  "zh"
+#define SVLIBRARY( Base ) SAL_MODULENAME( Base DLL_FNAME_POSTFIX )
 #include <osl/module.h>
 #endif

diff --git a/sal/textenc/textenc.cxx b/sal/textenc/textenc.cxx
index 7c9592c2e7a8..4ea6229ea3be 100644
--- a/sal/textenc/textenc.cxx
+++ b/sal/textenc/textenc.cxx
@@ -41,6 +41,8 @@
 #include "tcvtutf8.hxx"
 #include "tenchelp.hxx"
 
+#define SAL_TEXTENC_MODULE SAL_MODULENAME( "sal_textenc" DLL_FNAME_POSTFIX )
+ 
 #define NOTABUNI_START 0xFF
 #define NOTABUNI_END 0x00
 
@@ -381,7 +383,9 @@ void thisModule() {}
 class FullTextEncodingData {
 public:
     FullTextEncodingData() {
-        if (!module_.loadRelative(&thisModule, SAL_MODULENAME("sal_textenclo")))
+
+
+        if (!module_.loadRelative(&thisModule, SAL_TEXTENC_MODULE ))
         {
             SAL_WARN( "sal.textenc", "Loading sal_textenc library failed" );
             std::abort();
diff --git a/vcl/source/app/salplug.cxx b/vcl/source/app/salplug.cxx
index 1868853e93b8..aa05f0b44514 100644
--- a/vcl/source/app/salplug.cxx
+++ b/vcl/source/app/salplug.cxx
@@ -73,7 +73,7 @@ SalInstance* tryInstance( const OUString& rModuleBase, bool bForce = false )
 #ifdef SAL_DLLPREFIX
             SAL_DLLPREFIX
 #endif
-            "vclplug_" + aUsedModuleBase + "lo" SAL_DLLEXTENSION );
+            "vclplug_" + aUsedModuleBase + DLL_FNAME_POSTFIX SAL_DLLEXTENSION );
 
     osl::Module aMod;
     if (aMod.loadRelative(reinterpret_cast<oslGenericFunction>(&tryInstance), aModule, SAL_LOADMODULE_GLOBAL))
@@ -293,7 +293,7 @@ void DestroySalInstance( SalInstance *pInst )
 void SalAbort( const OUString& rErrorText, bool bDumpCore )
 {
     if( rErrorText.isEmpty() )
-        std::fprintf( stderr, "Application Error\n" );
+        std::fprintf( stderr, "Application Error - SalAbort\n" );
     else
     {
         CrashReporter::addKeyValue("AbortMessage", rErrorText, CrashReporter::Write);
diff --git a/vcl/source/window/builder.cxx b/vcl/source/window/builder.cxx
index 884a36474a5c..93c279ccad05 100644
--- a/vcl/source/window/builder.cxx
+++ b/vcl/source/window/builder.cxx
@@ -69,6 +69,10 @@
 #include <wizdlg.hxx>
 #include <tools/svlibrary.h>
 
+#include <string>
+#define concatc_str(first, second) (std::string(first) + std::string(second)).c_str()
+
+
 #if defined(DISABLE_DYNLOADING) || defined(LINUX)
 #include <dlfcn.h>
 #endif
@@ -1693,11 +1697,11 @@ void VclBuilder::preload()
 // find -name '*ui*' | xargs grep 'class=".*lo-' |
 //     sed 's/.*class="//' | sed 's/-.*$//' | sort | uniq
     static const char *aWidgetLibs[] = {
-        "sfxlo",  "svtlo",     "svxcorelo", "foruilo",
-        "vcllo",  "svxlo",     "cuilo",     "swlo",
-        "swuilo", "sclo",      "sdlo",      "chartcontrollerlo",
-        "smlo",   "scuilo",    "basctllo",  "sduilo",
-        "scnlo",  "xsltdlglo", "pcrlo" // "dbulo"
+        concatc_str("sfx", DLL_FNAME_POSTFIX), concatc_str("svt", DLL_FNAME_POSTFIX), concatc_str("svxcore", DLL_FNAME_POSTFIX), concatc_str("forui", DLL_FNAME_POSTFIX),
+        concatc_str("vcl", DLL_FNAME_POSTFIX), concatc_str("svx", DLL_FNAME_POSTFIX), concatc_str("cui", DLL_FNAME_POSTFIX),  concatc_str("sw", DLL_FNAME_POSTFIX),
+        concatc_str("swuis", DLL_FNAME_POSTFIX), concatc_str("sc", DLL_FNAME_POSTFIX), concatc_str("sd", DLL_FNAME_POSTFIX), concatc_str("chartcontroller", DLL_FNAME_POSTFIX),
+        concatc_str("sm", DLL_FNAME_POSTFIX), concatc_str("scui", DLL_FNAME_POSTFIX), concatc_str("basctl", DLL_FNAME_POSTFIX), concatc_str("sdui", DLL_FNAME_POSTFIX),
+        concatc_str("scn", DLL_FNAME_POSTFIX), concatc_str("xsltdlg", DLL_FNAME_POSTFIX), concatc_str("pcr", DLL_FNAME_POSTFIX) // "dbulo"
     };
     for (const auto & lib : aWidgetLibs)
     {
@@ -1716,7 +1720,7 @@ extern "C" VclBuilder::customMakeWidget lo_get_custom_widget_func(const char* na
 
 namespace
 {
-// Takes a string like "sfxlo-SidebarToolBox"
+// Takes a string like "sfxzh-SidebarToolBox"
 VclBuilder::customMakeWidget GetCustomMakeWidget(const OString& name)
 {
     VclBuilder::customMakeWidget pFunction = nullptr;
@@ -1757,7 +1761,8 @@ VclBuilder::customMakeWidget GetCustomMakeWidget(const OString& name)
                         assert(ok && "couldn't even directly dlsym the sFunction (available via preload)");
                     }
 #endif
-                    assert(ok && "bad module name in .ui");
+                    SAL_WARN("vcl", "modul " << sModule);
+                    assert(ok && "bad module name in .ui ");
                 }
                 else
                 {
@@ -2345,7 +2350,7 @@ VclPtr<vcl::Window> VclBuilder::makeObject(vcl::Window *pParent, const OString &
     }
     else if(name == "NotebookBarAddonsToolMergePoint")
     {
-        customMakeWidget pFunction = GetCustomMakeWidget("sfxlo-NotebookbarToolBox");
+        customMakeWidget pFunction = GetCustomMakeWidget("sfxzh-NotebookbarToolBox");
         if(pFunction != nullptr)
             NotebookBarAddonsMerger::MergeNotebookBarAddons(pParent, pFunction, m_xFrame, *m_pNotebookBarAddonsItem, rMap);
         return nullptr;
diff --git a/vcl/win/app/salinst.cxx b/vcl/win/app/salinst.cxx
index 5a10de216f96..72729ec686b0 100644
--- a/vcl/win/app/salinst.cxx
+++ b/vcl/win/app/salinst.cxx
@@ -83,7 +83,7 @@ void SalAbort( const OUString& rErrorText, bool )
     {
         // make sure crash reporter is triggered
         RaiseException( 0, EXCEPTION_NONCONTINUABLE, 0, nullptr );
-        FatalAppExitW( 0, L"Application Error" );
+        FatalAppExitW( 0, L"Application Error - SalAbort vcl/win/app" );
     }
     else
     {
