diff --git a/desktop/win32/source/applauncher/launcher.cxx b/desktop/win32/source/applauncher/launcher.cxx
index bf80dba2cc5a..9c7937741123 100644
--- a/desktop/win32/source/applauncher/launcher.cxx
+++ b/desktop/win32/source/applauncher/launcher.cxx
@@ -50,7 +50,7 @@ extern "C" int APIENTRY wWinMain( HINSTANCE, HINSTANCE, LPWSTR, int )
 
     GetModuleFileNameW( nullptr, szApplicationName, MAX_PATH );
     _wsplitpath( szApplicationName, szDrive, szDir, szFileName, szExt );
-    _wmakepath( szApplicationName, szDrive, szDir, L"soffice", L".exe" );
+    _wmakepath( szApplicationName, szDrive, szDir, SOFFICE_EXE_NAME, L".exe" );
 
     PROCESS_INFORMATION aProcessInfo;
 
diff --git a/desktop/win32/source/applauncher/launcher.hxx b/desktop/win32/source/applauncher/launcher.hxx
index e28cc7466e43..8b5688465359 100644
--- a/desktop/win32/source/applauncher/launcher.hxx
+++ b/desktop/win32/source/applauncher/launcher.hxx
@@ -20,6 +20,7 @@
 #pragma once
 
 #define WIN32_LEAN_AND_MEAN
+#define SOFFICE_EXE_NAME L"zoffice"
 #include <windows.h>
 
 extern WCHAR APPLICATION_SWITCH[];
 